builtin/providers/aws/resource_aws_elasticsearch_domain.go

```diff
@@ -256,6 +256,11 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 		DomainName: aws.String(d.Get("domain_name").(string)),
 	})
 	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "RecordNotFoundException" {
```

Comment:
Are you sure this is the right error code?

http://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-configuration-api.html#es-configuration-api-errors-resourcenotfound

the API reference says "ResourceNotFoundException", i.e. `Resource` not `Record`